### PR TITLE
Update identity web templates

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Fixes https://github.com/dotnet/aspnetcore/issues/25810 and updates identity.web version.

### CODE REVIEW

**Comment by JunTaoLuo** [comment]
FYI @Pilchie for rc2. 

@jmprieur in case you want to do some verification. I haven't tested anything yet but I'll at least try generating the templates some time in the evening.

File: src/ProjectTemplates/Web.ProjectTemplates/BlazorServerWeb-CSharp.csproj.in
Comment (dougbu) [inline src/ProjectTemplates/Web.ProjectTemplates/BlazorServerWeb-CSharp.csproj.in:27]: Why isn't this `Microsoft.Identity.Web.MicrosoftGraph`❔ Same in the other `*.in` files.

File: eng/Dependencies.props
Comment (dougbu) [inline eng/Dependencies.props:171]: Why are these needed❔ I don't see `@(Reference)` items using either.

File: src/ProjectTemplates/Web.ProjectTemplates/BlazorServerWeb-CSharp.csproj.in
Comment (JunTaoLuo) [reply src/ProjectTemplates/Web.ProjectTemplates/BlazorServerWeb-CSharp.csproj.in:27]: Typo

File: eng/Dependencies.props
Comment (JunTaoLuo) [reply eng/Dependencies.props:171]: Ah force of habit. Right these are only used in the template

**Comment by JunTaoLuo** [comment]
I took a look at the generated templates and they look right to me. @jmprieur can you also verify them?

**Comment by Pilchie** [comment]
Approved pending review signoff, and ideally @jmprieur taking a look as well.

File: src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Server/Controllers/WeatherForecastController.cs
Comment (jmprieur) [inline src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Server/Controllers/WeatherForecastController.cs:39]: Do we need GenerateApiOrGraph? even if the Web API does not call anything the scope needs to be verified?

File: src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-CSharp/Controllers/WeatherForecastController.cs
Comment (jmprieur) [inline src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-CSharp/Controllers/WeatherForecastController.cs:38]: Same question. Do we need GenerateApiOrGraph ?

**Review by jmprieur — COMMENTED** [review]
LGTM, @JunTaoLuo 
I believe that one condition is not necessary (GenerateApiOrGraph) to generate the scopes in the Web APIs

File: src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Server/Controllers/WeatherForecastController.cs
Comment (JunTaoLuo) [reply src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Server/Controllers/WeatherForecastController.cs:39]: That's a valid question, though I'm not sure if I should be making that call. When I wrote the condition, I simply looked at where `scopeRequiredByApi` was used. In this case, I'm trying to accommodate the calls on line 58, 95 and 117. Are you saying that we should consider removing lines 58 and 95?

Comment (jmprieur) [reply src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Server/Controllers/WeatherForecastController.cs:39]: No, we should not remove 58 and 95, @JunTaoLuo. It's just that 
`GenerateApiOrGraph` implies `OrganizationalAuth` or `IndividualB2CAuth` (you can only generate API in the case of  `OrganizationalAuth` or `IndividualB2CAuth` and you can only generate Graph in the case of  `OrganizationalAuth`), and therefore `GenerateApiOrGraph` in line 39 is redundant (but not harmful)

**Review by jmprieur — APPROVED** [review]
LGTM.

File: src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Server/Controllers/WeatherForecastController.cs
Comment (JunTaoLuo) [reply src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Server/Controllers/WeatherForecastController.cs:39]: Ah, I misunderstood what you said, and I see what you mean now. Yea, this can be simplified.

**Pull request merged**
